parser/src/parser/grammar.lalrpop

```diff
@@ -515,8 +517,12 @@ Range: Range = {
     <start: Num_u64> ".." <end: Num_u64> => Range::new(start as usize, end as usize)
 }
 
-RandArrayAccess: Identifier = {
-    "$" <n:identifier> => Identifier(format!("${n}"))
+DeclIdentRef: Identifier = {
+    <n:decl_ident_ref> => Identifier(n.to_string())
```

Comment:
minor nit: I was wondering if we should change the name from `n` here and elsewhere to something else because it sounds like it could be referring to a number. In some places we use `ident`, so we could use that. This could be done in another PR but I was just thinking about changing this earlier but forgot to.

Reply:
Never mind, I changed it now. There were only a couple places :)